samples/add-features-with-contingent-values/src/main/java/com/esri/arcgismaps/sample/addfeatureswithcontingentvalues/components/MapViewModel.kt

```diff
@@ -0,0 +1,380 @@
+        }
+
+        return sliderControlParameters
+    }
+
+    fun onBufferSizeSelect(bufferSize: Int) {
+        feature?.attributes?.set("BufferSize", bufferSize)
+        _uiState.value = _uiState.value.copy(buffer = bufferSize)
+    }
+
+    /**
+     * Ensure that the selected values are a valid combination.
+     * If contingencies are valid, then display [feature] on the [mapPoint]
+     */
+    fun validateContingency(mapPoint: Point) {
+        // check if all the features have been set
+        if (featureTable == null) {
+            messageDialogVM.showMessageDialog("Input all values to add a feature to the map")
+            return
+        }
+
+        // validate the feature's contingencies
+        val contingencyViolations = feature?.let {
+            featureTable?.validateContingencyConstraints(it)
+        } ?: return messageDialogVM.showMessageDialog("No feature attribute was selected")
+
+        // if there are no contingency violations
+        if (contingencyViolations.isEmpty()) {
+            // the feature is valid and ready to add to the feature table
+            // create a symbol to represent a bird's nest
+            val symbol = SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, Color.black, 11F)
+            // add the graphic to the graphics overlay
+            graphicsOverlay.graphics.add(Graphic(mapPoint, symbol))
+
+            // set the geometry of the feature to the map point
+            feature?.geometry = mapPoint
+
+            // create the graphic of the feature
+            val graphic = feature?.let { createGraphic(it) }
+            // add the graphic to the graphics overlay
+            graphic?.let { graphicsOverlay.graphics.add(it) }
+
+            // add the feature to the feature table
+            viewModelScope.launch {
+                feature?.let { featureTable?.addFeature(it) }
+                feature?.load()?.getOrElse {
+                    return@launch messageDialogVM.showMessageDialog("Error", it.message.toString())
+                }
+            }
+        } else {
+            val violations = contingencyViolations.map { violation -> violation.fieldGroup.name }
+                .joinToString(separator = "\n")
+            messageDialogVM.showMessageDialog(
+                "Invalid contingent values",
+                "${contingencyViolations.size} violations found:\n" + violations
+            )
+        }
+
+    }
+
+    /**
+     * Clears feature and attributes held in the view model to avoid inconsistent state
+     * after feature is created, fails to create, etc.
+     */
+    fun clearFeature() {
+        feature = null
+        _uiState.value = UIState(null, null, 0)
+        _sliderControlParameters.value = SliderControlParameters(false, 0, 0)
+    }
+
+}
+
+/**
+ *  Enable status, maximum, and minimum values for the buffer size slider
+ */
+data class SliderControlParameters(val isEnabled: Boolean, val minRange: Int, val maxRange: Int)
```

Comment:
Could use default parameters:
```suggestion
data class SliderControlParameters(val isEnabled: Boolean = false, val minRange: Int = 0, val maxRange: Int = 0)
```